source/MRPch/MRSuppressWarning.h

```diff
@@ -0,0 +1,23 @@
+#pragma once
+
+#if defined( __clang__ )
+#define SUPPRESS_DEPRECATION_WARNING( x ) \
+_Pragma( "clang diagnostic push" )      \
+_Pragma( "clang diagnostic ignored \"-Wdeprecated-declarations\"" ) \
+{ x }                                   \
+_Pragma( "clang diagnostic pop" )
+#elif defined( __GNUC__ )
+#define SUPPRESS_DEPRECATION_WARNING( x ) \
+_Pragma( "GCC diagnostic push" )        \
+_Pragma( "GCC diagnostic ignored \"-Wdeprecated-declarations\"" ) \
+{ x }                                   \
+_Pragma( "GCC diagnostic pop" )
+#elif defined( _MSC_VER )
+#define SUPPRESS_DEPRECATION_WARNING( x ) \
+_pragma( warning( push ) )              \
+_pragma( warning( disable: 4996 ) )     \
+{ x }                                   \
+_pragma( warning( pop ) )
+#else
+#define SUPPRESS_DEPRECATION_WARNING( x ) { x }
+#endif
```

Comment:
better add separate PUSH and POP macros, and let them have warning in argument

Reply:
also lets use it in other places where we suppress warnings

Reply:
Clang/GCC and MSVC have different warning option formats, unified macros will have to have two arguments to support both of them.